File: concept/17_recursion/string/check_palindrome/check_palindrome.dart
bool isPalindrome(String s, int i, int end) {
  if (i > end) {
    return true;
  }

  if (s[i] != s[end]) {
    return false;
  } else {
    // i++;
    // end--;
    return isPalindrome(s, i + 1, end - 1);
  }
}

void main() {
  String s = "MOM";

  bool v = isPalindrome(s, 0, s.length - 1);

  if (v) {
    print("It's Palindrome");
  } else {
    print("It not Palindrome");
  }
}

File: concept/17_recursion/string/check_palindrome/check_palindrome.cpp
#include <iostream>
using namespace std;

bool isPalindrome(string s, int i, int end)
{

    if (i > end)
    {
        return true;
    }

    if (s[i] != s[end])
    {
        return false;
    }
    else
    {
        // i++;
        // end--;
        return isPalindrome(s, i + 1, end - 1);
    }
}

int main()
{
    string s = "MOM";

    bool is = isPalindrome(s, 0, s.length() - 1);

    if (is)
    {
        cout << "It's Palindrome" << endl;
    }
    else
    {
        cout << "It not Palindrome" << endl;
    }
}